### PR TITLE
Replace timestamp with typesafe wrapper

### PR DESCRIPTION
**Pull request by i1i1**
Signed-off-by: i1i1 <vanyarybin1@live.ru>

### CODE REVIEW

**Review by humb1t — COMMENTED** [review]
Why can't we use DateTime directly?

File: iroha_data_model/src/query.rs [hunk 1]
```diff
@@ -78,14 +79,8 @@ pub struct QueryResult(pub Value);
...
     /// Constructs a new request with the `query`.
     pub fn new(query: QueryBox) -> Self {
-        QueryRequest {
-            timestamp: SystemTime::now()
-                .duration_since(SystemTime::UNIX_EPOCH)
-                .expect("Failed to get System Time.")
-                .as_millis()
-                .to_string(),
-            query,
-        }
```
Comment (eadventurous) [inline iroha_data_model/src/query.rs:88]: I think the timestamp that we had here is okay, as it is a unix timestamp and therefore it is universal. We just don't want to convert it to string. Use u64 as in `creation_time` in transaction.

Comment (eadventurous) [reply iroha_data_model/src/query.rs:88]: Maybe we can use scale's `CompactEncoding` for this, but I am not sure on this.

**Review by eadventurous — CHANGES_REQUESTED** [review]

File: iroha_data_model/src/query.rs [hunk 2]
```diff
@@ -78,27 +77,28 @@ pub struct QueryResult(pub Value);
...
+            timestamp_ms,
             query,
         }
     }
 
     /// `Hash` of this request.
     pub fn hash(&self) -> Hash {
         let mut payload: Vec<u8> = self.query.clone().into();
-        payload.extend_from_slice(self.timestamp.as_bytes());
+        payload.extend_from_slice(&self.timestamp_ms.0.to_le_bytes());
```
Comment (eadventurous) [inline iroha_data_model/src/query.rs:94]: We have a rule in our style guide not to access inner elements by position index, please destructure it instead.

File: iroha_data_model/src/query.rs [hunk 3]
```diff
@@ -53,9 +53,8 @@ pub enum QueryBox {
 /// I/O ready structure to send queries.
 #[derive(Debug, Io, Encode, Decode, Clone)]
 pub struct QueryRequest {
-    //TODO: why use `String` for timestamp? maybe replace it with milliseconds in `u64`
     /// Timestamp of the query creation.
-    pub timestamp: String,
+    pub timestamp_ms: Compact<u128>,
```
Comment (eadventurous) [inline iroha_data_model/src/query.rs:57]: Can we use `#[codec(compact)]` attribute here instead?

Comment (eadventurous) [reply iroha_data_model/src/query.rs:57]: This way it will be easier to use `QueryRequest.timestamp_ms` in code.

**Review by eadventurous — CHANGES_REQUESTED** [review]

**Review by eadventurous — APPROVED** [review]

**Pull request merged**